src/core/datatypes/trees/Tree.cpp

```diff
@@ -431,6 +431,10 @@ void Tree::executeMethod(const std::string &n, const std::vector<const DagNode *
     if ( n == "parent" )
     {
         long index = static_cast<const TypedDagNode<long> *>( args[0] )->getValue()-1;
+        if (getNode( index ).isRoot() == true)
+        {
+            throw RbException("Root has no parent.");
```

Comment:
If we throw an error like this, a script might stop. If we return -1, then the script could handle this. Not sure what's better/safer.

Reply:
Hmm... if you return `-1`, is there a chance the script would continue and yield non-sensical results?

It looks like if you call `getNode( -1 )` revbayes might segfault:

```
const TopologyNode& Tree::getNode(size_t idx) const
{

    if ( idx >= nodes.size() )
    {
        throw RbException("Index out of bounds in getNode.");
    }

    return *nodes[idx];
}
```

I think we should throw an exception instead of returning `-1`.  And probably we should throw an exception also if the user does `getNode(-1)`.

Reply:
There aren't very many places in RevBayes that use `-1` to indication "no value".  So maybe its best for scripts to handle this situation first checking whether a node is the root node before computing its parent...